### PR TITLE
[9News] Add Extractor

### LINKED ISSUE #8831: 9news.com.au site support
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

Australia

### Example URLs

- Single video: https://www.9news.com.au/videos/national/fair-trading-pulls-dozens-of-toys-from-shelves/clqgc7dvj000y0jnvfism0w5m

### Provide a description that is worded well enough to be understood

I'm not really very good with programming, so I may not be much help, but support for this site would be very nice :)

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://www.9news.com.au/videos/national/fair-trading-pulls-dozens-of-toys-from-shelves/clqgc7dvj000y0jnvfism0w5m']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2023.12.21.232720 from yt-dlp/yt-dlp-nightly-builds [632b8ee54] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 2023-01-09-git-13d04e30d7-full_build-www.gyan.dev (setts), ffprobe 2023-01-09-git-13d04e30d7-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.19.0, brotli-1.1.0, certifi-2023.11.17, mutagen-1.47.0, requests-2.31.0, sqlite3-3.35.5, urllib3-2.1.0, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1796 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2023.12.21.232720 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2023.12.21.232720 from yt-dlp/yt-dlp-nightly-builds)
[generic] Extracting URL: https://www.9news.com.au/videos/national/fair-trading-pulls-dozens-of-toys-from-shelves/clqgc7dvj000y0jnvfism0w5m
[generic] clqgc7dvj000y0jnvfism0w5m: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] clqgc7dvj000y0jnvfism0w5m: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://www.9news.com.au/videos/national/fair-trading-pulls-dozens-of-toys-from-shelves/clqgc7dvj000y0jnvfism0w5m
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1581, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1716, in __extract_info
  File "yt_dlp\extractor\common.py", line 718, in extract
  File "yt_dlp\extractor\generic.py", line 2531, in _real_extract
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.9news.com.au/videos/national/fair-trading-pulls-dozens-of-toys-from-shelves/clqgc7dvj000y0jnvfism0w5m
```

### PR DESCRIPTION
**Pull request by SirElderling**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Initial video extractor for the ``9News`` website.

Fixes #8831 

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [X] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by seproDev** [comment]
I think you should just be able to use the Brightcove extractor:
```python
return self.url_result(
    f'https://players.brightcove.net/{account}/default_default/index.html?videoId={video_id}',
    BrightcoveNewIE, video_id)
```

**Comment by SirElderling** [comment]
> I think you should just be able to use the Brightcove extractor:
> 
> ```python
> return self.url_result(
>     f'https://players.brightcove.net/{account}/default_default/index.html?videoId={video_id}',
>     BrightcoveNewIE, video_id)
> ```

File: yt_dlp/extractor/news9.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        info = traverse_obj(initial_state, {
+            'video_id': ('videoIndex', 'currentVideo', 'brightcoveId'),
+            'account': ('videoIndex', 'config', 'account'),
+        }) or traverse_obj(initial_state, {
+            'video_id': ('article', ..., 'media', lambda _, v: v['type'] == 'video', 'urn'),
+            'account': ('videoIndex', 'config', 'video', 'account')}, get_all=False)
+
+        video_id = info.get('video_id')
+        account = info.get('account')
```
Comment (seproDev) [inline yt_dlp/extractor/news9.py:69]: How about like so:
```suggestion
        video_id = traverse_obj(
            initial_state, ('videoIndex', 'currentVideo', 'brightcoveId'),
            ('article', ..., 'media', lambda _, v: v['type'] == 'video', 'urn'), get_all=False)
        account = traverse_obj(initial_state, ('videoIndex', 'config', (None, 'video'), 'account'), get_all=False)
```

File: yt_dlp/extractor/news9.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            'upload_date': '20231222',
+            'uploader_id': '664969388001',
+            'tags': ['networkclip', 'aunews_aunationalninenews', 'presents', 'teachers', 'kindergarten', 'au_news'],
+        },
+    }]
+
+    def _real_extract(self, url):
+        article_id = self._match_id(url)
+        webpage = self._download_webpage(url, article_id)
+        initial_state = self._search_json(r'var\s__INITIAL_STATE__\s*=', webpage, 'initial state', article_id)
```
Comment (seproDev) [inline yt_dlp/extractor/news9.py:59]: Might as well:
```suggestion
        initial_state = self._search_json(r'var\s+__INITIAL_STATE__\s*=', webpage, 'initial state', article_id)
```

**Review by seproDev — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/news9.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+from .common import InfoExtractor
+from .brightcove import BrightcoveNewIE
+
+from ..utils import ExtractorError, traverse_obj
+
+
+class News9IE(InfoExtractor):
```
Comment (seproDev) [inline yt_dlp/extractor/news9.py:7]: Should we name the extractor `NineNewsIE` instead? That's also what [Wikipedia](https://en.wikipedia.org/wiki/Nine_News) calls the outlet.

File: yt_dlp/extractor/news9.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            'upload_date': '20231222',
+            'uploader_id': '664969388001',
+            'tags': ['networkclip', 'aunews_aunationalninenews', 'presents', 'teachers', 'kindergarten', 'au_news'],
+        },
+    }]
+
+    def _real_extract(self, url):
+        article_id = self._match_id(url)
+        webpage = self._download_webpage(url, article_id)
+        initial_state = self._search_json(r'var\s+__INITIAL_STATE__\s*=', webpage, 'initial state', article_id)
```
Comment (seproDev) [inline yt_dlp/extractor/news9.py:59]:
```suggestion
        initial_state = self._search_json(
            r'var\s+__INITIAL_STATE__\s*=', webpage, 'initial state', article_id)
```

File: yt_dlp/extractor/news9.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        video_id = traverse_obj(
+            initial_state, ('videoIndex', 'currentVideo', 'brightcoveId'),
+            ('article', ..., 'media', lambda _, v: v['type'] == 'video', 'urn'), get_all=False)
+        account = traverse_obj(initial_state, ('videoIndex', 'config', (None, 'video'), 'account'), get_all=False)
```
Comment (seproDev) [inline yt_dlp/extractor/news9.py:63]:
```suggestion
        account = traverse_obj(initial_state, (
            'videoIndex', 'config', (None, 'video'), 'account'), get_all=False)
```

**Review by seproDev — APPROVED** [review]

File: yt_dlp/extractor/news9.py (hunk 3 above)
Comment (SirElderling) [reply yt_dlp/extractor/news9.py]: Thanks for the suggestion. Makes more sense than reversing the name.

**Review by bashonly — APPROVED** [review]

**Pull request merged**